src/sagemaker/user_agent.py

```diff
@@ -15,9 +15,18 @@
 
 import platform
 import sys
+import json
+import os
 
 import importlib_metadata
 
+SDK_PREFIX = "AWS-SageMaker-Python-SDK"
+STUDIO_PREFIX = "AWS-SageMaker-Studio"
+NOTEBOOK_PREFIX = "AWS-SageMaker-Notebook-Instance"
+
+NOTEBOOK_METADATA_FILE = "/etc/opt/ml/sagemaker-notebook-instance-version.txt"
```

Comment:
Are we sure these files exists in all the notebook instances? 

If yes, how we are planning to ensure that when this contract is broken, we able to catch it before than later?

Reply:
Yes, I verified that this file exists. Currently, we don't have a way to track if this contract is broken since that source of this information is not open. We will manually have to update it here if that changes in future. 
I assume that this metadata will NOT change since these are base assumptions for multiple developments.
<img width="580" alt="Screenshot 2024-02-22 at 4 34 29 PM" src="https://github.com/aws/sagemaker-python-sdk/assets/110067132/a95986f9-fc55-464d-ba82-68fe16396292">

Reply:
`/opt/ml/metadata/resource-metadata.json` should be the same for Classic Notebooks and Studio can we use that here?

https://docs.aws.amazon.com/sagemaker/latest/dg/nbi-metadata.html
https://docs.aws.amazon.com/en_en/sagemaker/latest/dg/notebooks-run-and-manage-metadata.html

For backwards compatibility the file should always exist, I believe

Reply:
Yes confirmed that both of these files will exists and should not change for backwards compatibility.
However, as per below screenshot `/etc/opt/ml/sagemaker-notebook-instance-version.txt` file emits the notebook-instance version used whereas `/opt/ml/metadata/resource-metadata.json` has only `ResourceArn` which is not of our interest to capture platform and its version in the logs. Hence we should continue to use the existing file for logging.
![Screenshot 2024-02-26 at 3 11 34 PM](https://github.com/aws/sagemaker-python-sdk/assets/110067132/b6018bb4-fed3-4e91-a256-108fb506e927)